src/include/common/concurrent_pointer_vector.h

```diff
@@ -0,0 +1,280 @@
+#pragma once
+
+#include <common/constants.h>
+#include <common/macros.h>
+#include <common/shared_latch.h>
+#include <execution/util/execution_common.h>
+#include <tbb/mutex.h>
+#include <atomic>              // NOLINT
+#include <condition_variable>  // NOLINT
+
+namespace terrier::common {
+
+/**
+ * A concurrent Vector of pointers that supports inserts and accesses at an index
+ */
+template <class T>
+class alignas(Constants::CACHELINE_SIZE) ConcurrentPointerVector {
+ public:
+  /**
+   * ConcurrentPointerVector constuctor
+   *
+   * @param start_size optional initial allocation size for the vector. Default is START_SIZE
+   */
+  explicit ConcurrentPointerVector(uint64_t start_size = START_SIZE)
```

Comment:
Andy is generally not in favor of default arguments, and in this case the default argument seems to be a little fishy.

Could you TERRIER_ASSERT that start size shouldn't be 0 instead?

Reply:
done